spec/controllers/spree/admin/payment_methods_controller_spec.rb

```diff
@@ -8,6 +8,59 @@ class GatewayWithPassword < PaymentMethod
   end
 
   describe Admin::PaymentMethodsController, type: :controller do
+    let(:user) {
+      create(:user, enterprises: [create(:distributor_enterprise)])
+    }
+
+    describe "#new" do
+      before { allow(controller).to receive(:spree_current_user) { user } }
+
+      it "allows to select from a range of payment gateways" do
+        spree_get :new
+        providers = assigns(:providers).map(&:to_s)
```

Comment:
When Rails reloads classes, the reloaded class has a different object id to the original. The original providers are stored at boot time and not reloaded properly and therefore we can't compare the to reloaded providers. I'm converting the names to strings and compare those instead.